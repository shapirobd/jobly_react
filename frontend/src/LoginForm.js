import React from "react";

const LoginForm = ({ handleSubmit, handleChange }) => {
	return (
		<form className="Login-form" onSubmit={handleSubmit}>
			<label htmlFor="username">Username</label>
			<input
				type="text"
				onChange={handleChange}
				name="username"
				id="username"
			></input>
			<label htmlFor="password">Password</label>
			<input
				type="password"
				onChange={handleChange}
				name="password"
				id="password"
			></input>
			<div className="Login-btn-wrapper">
				<button type="submit" className="Login-btn">
					Submit
				</button>
			</div>
		</form>
	);
};
export default LoginForm;
