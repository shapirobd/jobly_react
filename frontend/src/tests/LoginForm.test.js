import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "../LoginForm";

test("smoke test", () => {
	render(
		<MemoryRouter>
			<LoginForm />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<LoginForm />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

test("contains correct inputs/labels", () => {
	const { getByLabelText, getByText } = render(
		<MemoryRouter>
			<LoginForm />
		</MemoryRouter>
	);
	expect(getByLabelText("Username")).toBeInTheDocument();
	expect(getByLabelText("Password")).toBeInTheDocument();
	expect(getByText("Submit")).toBeInTheDocument();
});
