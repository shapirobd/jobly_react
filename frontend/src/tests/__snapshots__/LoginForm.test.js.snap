// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <form
    class="Login-form"
  >
    <label
      for="username"
    >
      Username
    </label>
    <input
      id="username"
      name="username"
      type="text"
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      type="password"
    />
    <div
      class="Login-btn-wrapper"
    >
      <button
        class="Login-btn"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
